tests/unit_tests/sycl/propagator.sycl

```diff
@@ -0,0 +1,231 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <vecmem/memory/sycl/device_memory_resource.hpp>
+#include <vecmem/memory/sycl/shared_memory_resource.hpp>
+
+#include "propagator_kernel.hpp"
+#include "vecmem/utils/sycl/copy.hpp"
+
+// Simple asynchronous handler function
+auto handle_async_error = [](::sycl::exception_list elist) {
+    for (auto& e : elist) {
+        try {
+            std::rethrow_exception(e);
+        } catch (::sycl::exception& e) {
+            std::cout << "ASYNC EXCEPTION!!\n";
+            std::cout << e.what() << "\n";
+        }
+    }
+};
+
+class SyclPropagatorWithRkStepper
+    : public ::testing::TestWithParam<__plugin::vector3<scalar>> {};
+
+TEST_P(SyclPropagatorWithRkStepper, propagator) {
```

Comment:
Would it be possible to pool the host propagation part with the CUDA version somehow, e.g. in a separate header?

Reply:
Let me postpone this issue for now. I will do huge refactoring in the later PR regarding on this